torchaudio/prototype/tacotron2.py

```diff
@@ -0,0 +1,851 @@
+        postnet_kernel_size (int, optional): Postnet kernel size (Default: ``5``).
+        postnet_n_convolutions (int, optional): Number of postnet convolutions (Default: ``5``).
+        decoder_no_early_stopping (bool, optional): Stop decoding when all samples are finished (Default: ``False``).
+    """
+    def __init__(self,
+                 mask_padding: bool = False,
+                 n_mels: int = 80,
+                 n_symbols: int = 148,
+                 symbols_embedding_dim: int = 512,
+                 encoder_kernel_size: int = 5,
+                 encoder_n_convolutions: int = 3,
+                 encoder_embedding_dim: int = 512,
+                 attention_rnn_dim: int = 1024,
+                 attention_dim: int = 128,
+                 attention_location_n_filters: int = 32,
+                 attention_location_kernel_size: int = 31,
+                 n_frames_per_step: int = 1,
+                 decoder_rnn_dim: int = 1024,
+                 prenet_dim: int = 256,
+                 max_decoder_steps: int = 2000,
+                 gate_threshold: float = 0.5,
+                 p_attention_dropout: float = 0.1,
+                 p_decoder_dropout: float = 0.1,
+                 postnet_embedding_dim: int = 512,
+                 postnet_kernel_size: int = 5,
+                 postnet_n_convolutions: int = 5,
+                 decoder_no_early_stopping: bool = False) -> None:
+        super().__init__()
+
+        self.mask_padding = mask_padding
+        self.n_mels = n_mels
+        self.n_frames_per_step = n_frames_per_step
+        self.embedding = nn.Embedding(n_symbols, symbols_embedding_dim)
+        std = sqrt(2.0 / (n_symbols + symbols_embedding_dim))
+        val = sqrt(3.0) * std
+        self.embedding.weight.data.uniform_(-val, val)
+        self.encoder = _Encoder(encoder_n_convolutions,
+                                encoder_embedding_dim,
+                                encoder_kernel_size)
+        self.decoder = _Decoder(n_mels, n_frames_per_step,
+                                encoder_embedding_dim, attention_dim,
+                                attention_location_n_filters,
+                                attention_location_kernel_size,
+                                attention_rnn_dim, decoder_rnn_dim,
+                                prenet_dim, max_decoder_steps,
+                                gate_threshold, p_attention_dropout,
+                                p_decoder_dropout,
+                                not decoder_no_early_stopping)
+        self.postnet = _Postnet(n_mels, postnet_embedding_dim,
+                                postnet_kernel_size,
+                                postnet_n_convolutions)
+
+    def _parse_output(self,
+                      outputs: Tuple[Tensor, Tensor, Tensor, Tensor],
+                      output_lengths: Tensor) -> Tuple[Tensor, Tensor, Tensor, Tensor]:
```

Comment:
since this is a single purpose function, i'd recommend moving this code in place, with a comment.

Reply:
addressed in the new [commit](https://github.com/pytorch/audio/pull/1621/commits/5fccd01181b8a5d5aa3e33f0f0102c27bd154223).
Thanks.